### PR TITLE
[FAB-18315] Ch.Part.API: Add remove failure metric for chain status

### PR DESCRIPTION
**Pull request by stephyee**
#### Type of change

- New feature

#### Description
Adds StatusFailure metric for channel removal failures. Ensure GET List/:channelID returns correct ChannelInfo for pending channel removals. 

#### Related issues
[FAB-18315](https://jira.hyperledger.org/browse/FAB-18315)

### CODE REVIEW

File: orderer/common/multichannel/registrar.go [hunk 1]
```diff
@@ -719,11 +721,11 @@ func (r *Registrar) JoinChannel(channelID string, configBlock *cb.Block, isAppCh
...
 
-	if removalStatus, ok := r.pendingRemoval[channelID]; ok {
-		if removalStatus {
-			return types.ChannelInfo{}, types.ErrChannelPendingRemoval
-		} else {
+	if status, ok := r.pendingRemoval[channelID]; ok {
+		if status.Status == types.StatusFailure {
 			return types.ChannelInfo{}, types.ErrChannelRemovalFailure
+		} else {
+			return types.ChannelInfo{}, types.ErrChannelPendingRemoval
```
Comment (wlahti) [inline orderer/common/multichannel/registrar.go]: Style nit: no need for the else here.
```suggestion
		if status.Status == types.StatusFailure {
			return types.ChannelInfo{}, types.ErrChannelRemovalFailure
		} 
	       return types.ChannelInfo{}, types.ErrChannelPendingRemoval
```

File: orderer/common/types/channelinfo.go [hunk 2]
```diff
@@ -59,6 +59,7 @@ const (
 	StatusOnBoarding Status = "onboarding"
 	// The orderer is not storing any blocks for this channel.
 	StatusInactive Status = "inactive"
+	StatusFailure  Status = "failed"
```
Comment (wlahti) [inline orderer/common/types/channelinfo.go]: Add a comment here to explain the status? 

And I think we should have the constant name consistent with the value, so either StatusFailure = "failure" or StatusFailed = "failed". Not sure how I feel about "failed"... maybe "error" instead? Any thoughts @tock-ibm @jyellick ?

File: orderer/common/multichannel/registrar.go (hunk 1 above)
Comment (stephyee) [reply orderer/common/multichannel/registrar.go]: 🧹

File: orderer/common/types/channelinfo.go (hunk 2 above)
Comment (stephyee) [reply orderer/common/types/channelinfo.go]: For now, I went with StatusFailed. The comment you may hate.

**Review by yacovm — APPROVED** [review]

**Pull request merged**